gir/pkgconfig/pkgconfig.go

```diff
@@ -1,105 +1,152 @@
+		if ret[pkg] == "" {
+			badPkgs = append(badPkgs, pkg)
 		}
+	}
+	if len(badPkgs) > 0 {
+		aug, err := fn(badPkgs...)
+		if err != nil {
+			return nil, err
+		}
+		for pkg, val := range aug {
+			ret[pkg] = val
+		}
+	}
 
-		err := fs.WalkDir(os.DirFS(baseDir), ".",
-			func(path string, d fs.DirEntry, err error) error {
-				if errors.Is(err, fs.ErrPermission) {
-					return fs.SkipDir
-				}
-
-				if err != nil {
-					return err
-				}
-
-				if filepath.Ext(d.Name()) == ".gir" {
-					girFiles = append(girFiles, filepath.Join(baseDir, path))
-				}
-
-				return nil
-			},
-		)
+	return ret, nil
+}
 
+// AddPathSuffix takes a function that returns a map[pkg]dir and wraps
+// it so that each `dir` is set to `filepath.Join(dir, ...suffix)`.
+func AddPathSuffix(fn func(...string) (map[string]string, error), suffix ...string) func(...string) (map[string]string, error) {
+	return func(pkgs ...string) (map[string]string, error) {
+		ret, err := fn(pkgs...)
 		if err != nil {
-			return nil, fmt.Errorf("failed to walk %q: %w", baseDir, err)
+			return nil, err
 		}
+		for pkg, dir := range ret {
+			ret[pkg] = filepath.Join(append([]string{dir}, suffix...)...)
+		}
+		return ret, nil
 	}
+}
+
+// Prefix returns the install prefix for each requested package, or an
+// error if this cannot be determined for any of the packages.  Common
+// values are "/", "/usr", or "/usr/local".
+func Prefix(pkgs ...string) (map[string]string, error) {
+	return VariableOrElse("prefix", func(pkgs ...string) (map[string]string, error) {
+		return nil, fmt.Errorf("could not resolve install prefix for packages: %v", pkgs)
+	}, pkgs...)
+}
+
+// DataRootDir returns the directory for read-only
+// architecture-independent data files for each requested package, or
+// an error if this cannot be determined for any of the packages.  The
+// usual value is "${prefix}/share", i.e. "/usr/share" or
+// "/usr/local/share".
+func DataRootDir(pkgs ...string) (map[string]string, error) {
+	return VariableOrElse("datarootdir", AddPathSuffix(Prefix, "share"), pkgs...)
+}
+
+// DataDir returns the base directory for package-idiosyncratic
+// read-only architecture-independent data files for each requested
+// package, or an error if this cannot be determined for any of the
+// packages.  The usual value is "${datarootdir}", i.e. "/usr/share"
+// or "/usr/local/share"; this is *not* a per-package directory,
+// packages usually install their data to
+// "${datadir}/${package_name}/".
+func DataDir(pkgs ...string) (map[string]string, error) {
+	return VariableOrElse("datadir", DataRootDir, pkgs...)
+}
 
-	return girFiles, nil
+// GIRDir returns the directory for GObject Introspection Repository
+// XML files for each requested package, or an error if this cannot be
+// determined for any of the packages.  The usual value is
+// "${datadir}/gir-1.0", i.e. "/usr/share/gir-1.0" or
+// "/usr/local/share/gir-1.0".
+func GIRDir(pkgs ...string) (map[string]string, error) {
```

Comment:
Should `GIRDir`, `DataDir` and `DataRootDir` be named in their plural forms (`GIRDirs`, `DataDirs`, `DataRootDirs`) instead?

Reply:
In my mind, they'd be plural if there were multiple values per package. Like `libs` or `cppflags` or `targets`.

Put another way, these are doing the same as `Variable`, and naming it `Variables` feels clearly wrong.

But if you'd really prefer them to be plural, I can change them. Perhaps then I'd also rename `Variable` to `Values`?

Reply:
`Variable()` should ideally be called `PackagesVariable()`. The variable is singular, but you're operating on multiple packages.

`PackagesGIRDirectory` is quite long, but `GIRDirs` is more correct, I think.

Reply:
The more I thought about it, the more I liked making them plural and renaming `Variable()` to `Values()`. Done.